lib/endpoints/Queue.js

```diff
@@ -55,13 +55,14 @@ class Queue {
         action,
         maxInFlight,
         restoreRate,
-    }) {
+    }, deleteQueueFromSchedule) {
```

Comment:
What do you think if we had Queue extend EventEmitter, and emit an 'empty' event that the QueueScheduler listens for in register(), rather than passing in a new parameter here?

Reply:
I'm very curious, what do you think about delineating the classes even further in, for example, a QueueScheduler, Queue and QueueItem?

Reply:
well, let's do that EventEmitter then.  And isn't that what the current patch changes the classes to, having 3?

Reply:
Well it's a start, but the classes are not yet sufficiently disentangled. I wanted to shoot this idea by you, to see what you think. One of the things that we could do for example is to put the maxInFlight and restoreRate on the QueueScheduler. However, this is hard for me to do as I'm not familiar with some of the choices. 

To give an example:

````
if (this.inFlight >= this.maxInFlight * 0.5) {
	this.throttle();
}
````

Why are we multiplying the maxInFlight times 0.5?
I think that you thought very well about these choices (I don't doubt that they are necessary), but it's hard for someone not familiar with your reasoning to understand why these kind of choices were made. Therefore, I'm afraid that if I start working on disentangling everything further, I'll break the whole thing...

Reply:
I don't feel particularly strongly about that one way or the other, wrt which one has the maxInFlight and restoreRate, but it was put there as is, because it's directly associated to the Queue.  Queue keyed by sellerId/category/endpointname.  

It seemed prudent to assume that we're very likely to hit a throttle point for sure if we fill the queue halfway already.  As far as thought about that specific number goes, I'm pretty sure that i was still seeing a whole lot of requests get to Amazon that I didn't expect to, if I fired a few hundred rapidly, so I tweaked it down to assume that if we've reached the halfway point, we're probably going to fill the queue anyway.  Of course, there might also have been a fair amount of hitting hourly quota going rather than inflight quota while i was testing it.

I make no claims that it's the best way, only that it does a pretty decent job at not over-spamming the FBA servers.

:)

Reply:
I was out for a couple of weeks on a work trip.  Did you have it in mind to continue with some of the changes we've talked about here, or do it separately from this PR?

Reply:
Hi @ericblade! I didn't get around to doing that. I propose to that in a separate PR. I did work on extending the functionality of the package locally though. However, the code I wrote for doing that is not polished, I could create a new PR for that? Then you can check if you think the code is good enough to be pulled... If not, I'd totally understand.

Reply:
always glad to have a look.  i've been burning hard on a R&D project that is going to a demo phase this week, so haven't had any time to work on my own apps, or this.

Reply:
Don't worry, I know how it is. Good luck with the project!